operator/v1/types_ingress.go

```diff
@@ -0,0 +1,221 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// IngressController describes a managed ingress controller for the cluster. The
+// controller can service OpenShift Route and Kubernetes Ingress resources.
+//
+// When an IngressController is created, a new ingress controller deployment is
+// created to allow external traffic to reach the services that expose Ingress
+// or Route resources. Updating this resource may lead to disruption for public
+// facing network connections as a new ingress controller revision may be rolled
+// out.
+//
+// https://kubernetes.io/docs/concepts/services-networking/ingress-controllers
+//
+// Whenever possible, sensible defaults for the platform are used. See each
+// field for more details.
+type IngressController struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec is the specification of the desired behavior of the IngressController.
+	Spec IngressControllerSpec `json:"spec,omitempty"`
+	// status is the most recently observed status of the IngressController.
+	Status IngressControllerStatus `json:"status,omitempty"`
+}
+
+// IngressControllerSpec is the specification of the desired behavior of the
+// IngressController.
+type IngressControllerSpec struct {
+	// domain is a DNS name serviced by the ingress controller and is used to
+	// configure multiple features:
+	//
+	// * For the LoadBalancerService endpoint publishing strategy, domain is
+	//   used to configure DNS records. See endpointPublishingStrategy.
+	//
+	// * When using a generated default certificate, the certificate will be valid
+	//   for domain and its subdomains. See defaultCertificate.
+	//
+	// * The value is published to individual Route statuses so that end-users
+	//   know where to target external DNS records.
+	//
+	// domain must be unique among all IngressControllers, and cannot be
+	// updated.
+	//
+	// If empty, defaults to the cluster Ingress config domain value.
```

Comment:
@deads2k what's the best way to refer to the `ingress.config.openshift.io/cluster` resource here?

Reply:
fixed